scene/gui/base_button.cpp

```diff
@@ -144,7 +144,7 @@ void BaseButton::_toggled(bool p_pressed) {
 }
 
 void BaseButton::on_action_event(Ref<InputEvent> p_event) {
-	if (p_event->is_pressed()) {
+	if (p_event->is_pressed() && (!p_event->is_class("InputEventMouseButton") || status.hovering)) {
```

Comment:
A cast should be more performant than string comparisons, so I would suggest moving `Ref<InputEventMouseButton> mouse_button = p_event;` from below at the top of the method, and use that.